File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/C/s936984238_commented.c
#include <stdio.h>

int number;
int score[1000];

int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

int read() {
    if (scanf("%d", &number) == EOF) {
        return 0;
    }
    if (number == 0) {
        return 0;
    }
    for (int i = 0; i < number; i++) {
        scanf("%d", &score[i]);
    }
    return 1;
}

void slove() {
    qsort(score, number, sizeof(int), compare);
    int sum = 0;
    for (int i = 1; i < number - 1; i++) {
        sum += score[i];
    }
    printf("%d\n", sum / (number - 2));
}

int main() {
    while (read()) {
        slove();
    }
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Python/s936984238_commented.py
import sys

def main():
    sc = Scanner()
    while read(sc):
        solve()

def read(sc):
    number = sc.nextInt()
    if number == 0:
        return False
    global score
    score = [sc.nextInt() for _ in range(number)]
    return True

def solve():
    score.sort()
    sum = 0
    for i in range(1, len(score) - 1):
        sum += score[i]
    print(sum // (len(score) - 2))

class Scanner:
    def __init__(self):
        self.tokens = []
        self.token_index = 0
        self.input = sys.stdin

    def nextInt(self):
        if self.token_index == len(self.tokens):
            self.tokens = self.input.readline().strip().split()
            self.token_index = 0
        token = self.tokens[self.token_index]
        self.token_index += 1
        return int(token)

if __name__ == "__main__":
    main()  # End of Code
